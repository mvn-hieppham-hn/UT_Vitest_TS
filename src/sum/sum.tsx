export function sum(a: number, b: number) {
  return a + b;
}

export function sumOfSquares(k: number) {
  let sumOfSquares = 0;
  for (let i = 0; i < k + 1; i++) {
    sumOfSquares += i * i;
  }
  console.log("sumOfSquares:", sumOfSquares);
  return sumOfSquares;
}

export const factorial = (k: number) => {
  let factorialValue = 1;
  if (k === 0) {
    return 1;
  }
  for (let i = 1; i < k; i++) {
    factorialValue = factorialValue * i;
  }
  return factorialValue;
};

export const sumFactorial = (k: number) => {
  let sumFactorial = 0;
  if (k === 0) {
    return 1;
  }
  for (let i = 1; i <= k; i++) {
    sumFactorial += factorial(i);
  }
  return sumFactorial;
};

export const renderHTML = (html: string) => {
  return <h1>{html}</h1>;
};
