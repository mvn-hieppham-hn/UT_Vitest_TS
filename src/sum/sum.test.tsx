import { describe, expect, test } from "vitest";
import { renderHTML, sum, sumOfSquares } from "./sum";

describe("sum", () => {
  test("adds 1 + 2 to equal 3", () => {
    expect(sum(1, 2)).toBe(3);
  });

  test("adds 1^2 + 2^2 +3^2 to equal 14", () => {
    const totalSquares = sumOfSquares(7);
    expect(totalSquares).toMatchFileSnapshot("./test/text.html");
  });

  test("render html is 'Hello'", () => {
    const totalSquares = renderHTML("Hello hihi");
    expect(totalSquares).toMatchFileSnapshot("./test/basic.html");
  });
});
