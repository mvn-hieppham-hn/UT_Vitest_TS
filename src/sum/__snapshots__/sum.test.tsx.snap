// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sum > adds 1^2 + 2^2 +3^2 to equal 14 1`] = `140`;

exports[`sum > render html is 'Hello' 1`] = `
<h1>
  Hell
</h1>
`;
